backends/dpdk/dpdkMetadata.h

```diff
@@ -69,5 +69,17 @@ class DirectionToRegRead : public Transform {
     replaceDirectionWithRegRead(IR::IndexedVector<IR::DpdkAsmStatement> stmts);
 };
 
+class PrependPassRecircId : public Transform {
```

Comment:
The comment that accompanies this PR should be present in the code too, here.

Reply:
I am not familiar with PNA, can you also explain how this recirc works?

Reply:
The `recirculate() `extern corresponds to "_recirculate_" instruction and marks the packet for recirculation. This means that when packet reaches _tx_ instruction, it is reinjected into the pipeline for another full pass instead of being sent to the output port.

The same packet can be recirculated multiple times, with the recirculation pass number retrieved by the "_recircid_" instruction which corresponds to "_pass_" metadata. The pass number can be used by the program to execute different code on the same packet in different pipeline passes. The packet meta-data is preserved between the pipeline passes.

Reply:
DPDK reference patch for packet recirculation 
https://patches.dpdk.org/project/dpdk/patch/20220406185535.73084-6-cristian.dumitrescu@intel.com/